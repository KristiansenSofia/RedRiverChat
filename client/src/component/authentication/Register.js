import React, { Component } from 'react';

// Import NPM-modules
import TextField from 'material-ui/TextField';
import ImagesUploader from 'react-images-uploader';
import Button from 'material-ui/Button';
import Typography from 'material-ui/Typography';

// Import styles. newPasswordStyles for all imported components with a style attributes and CSS-file for classNames and id.
import {registerStyles} from "../../styles/AuthStyles";
import '../../styles/Styles.css'

/**
 *  Register-component.
 *
 *  @author Jimmy
 */


class Register extends Component {
    constructor(props){
        super(props);

        this.state = {
            userName: '',
            firstName: '',
            lastName: '',
            email: '',
            password: '',
            passwordConfirm: '',
            address: '',
            zipCode: '',
            city: '',
            personalIdentityNumber: '',
            telephoneNumber: '',
            nameRelative: '',
            image: [],
        };

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    /**
     *  Handle form-input. Input are added to this.state.
     *
     *  @author Jimmy
     */

    handleChange = name => event => {
        this.setState({
            [name]: event.target.value,
        });
    };

    /**
     *  Handle submit-button. Register-request is sent to server with form-input included.
     *
     *  @author Jimmy
     */

    handleSubmit() {
        this.props.openSnackBar('Välkommen ' + this.state.firstName + '!');
    }

    render() {
        return (
            <div className="Register">
                <Typography
                    variant="headline"
                    color="default"
                    align="left"
                    style={registerStyles.title}
                >
                    Registrera dig
                </Typography>
                <form style={registerStyles.container} noValidate autoComplete="off">
                    <TextField
                        id="name"
                        label="Användarnamn"
                        style={registerStyles.textField}
                        value={this.state.name}
                        onChange={this.handleChange('name')}
                        margin="normal"
                    />
                    <TextField
                        id="email"
                        label="Email"
                        style={registerStyles.textField}
                        value={this.state.email}
                        onChange={this.handleChange('email')}
                        margin="normal"
                        type="email"
                    />
                    <TextField
                        id="password"
                        label="Lösenord"
                        style={registerStyles.textField}
                        type="password"
                        autoComplete="current-password"
                        onChange={this.handleChange('password')}
                        margin="normal"
                    />
                    <TextField
                        id="passwordRepeat"
                        label="Bekräfta lösenord"
                        style={registerStyles.textField}
                        type="password"
                        autoComplete="current-password"
                        onChange={this.handleChange('passwordRepeat')}
                        margin="normal"
                    />
                    <TextField
                        id="firstName"
                        label="Förnamn"
                        style={registerStyles.textField}
                        value={this.state.firstName}
                        onChange={this.handleChange('firstName')}
                        margin="normal"
                    />
                    <TextField
                        id="lastName"
                        label="Efternamn"
                        style={registerStyles.textField}
                        value={this.state.lastName}
                        onChange={this.handleChange('lastName')}
                        margin="normal"
                    />
                    <TextField
                        id="address"
                        label="Adress"
                        style={registerStyles.textField}
                        value={this.state.address}
                        onChange={this.handleChange('address')}
                        margin="normal"
                    />
                    <TextField
                        id="zipCode"
                        label="Postnummer"
                        style={registerStyles.textField}
                        value={this.state.zipCode}
                        onChange={this.handleChange('zipCode')}
                        margin="normal"
                    />
                    <TextField
                        id="city"
                        label="Postort"
                        style={registerStyles.textField}
                        value={this.state.city}
                        onChange={this.handleChange('city')}
                        margin="normal"
                    />
                    <TextField
                        id="personaIdentityNumber"
                        label="Personnummer"
                        style={registerStyles.textField}
                        value={this.state.personalIdentityNumber}
                        onChange={this.handleChange('personalIdentityNumber')}
                        margin="normal"
                    />
                    <TextField
                        id="telephoneNumber"
                        label="Telefonnummer"
                        style={registerStyles.textField}
                        value={this.state.telephoneNumber}
                        onChange={this.handleChange('telephoneNumber')}
                        margin="normal"
                    />
                    <TextField
                        id="nameRelative"
                        label="Anhörigs namn"
                        style={registerStyles.textField}
                        value={this.state.nameRelative}
                        onChange={this.handleChange('nameRelative')}
                        margin="normal"
                    />
                    <ImagesUploader
                        url="http://example.com"
                        optimisticPreviews
                        multiple={false}
                        onLoadEnd={(err) => {
                            if (err) {
                                console.error(err);
                            }
                        }}
                        label="Välj profilbild"
                        styles={registerStyles.imageUpload}
                    />
                    <div className="RegisterButton">
                        <Button variant="raised" style={registerStyles.button} onClick={this.handleSubmit}>
                            Registrera
                        </Button>
                    </div>
                </form>
            </div>

        );
    }
}

export default Register;
