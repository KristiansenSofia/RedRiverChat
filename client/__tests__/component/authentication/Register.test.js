import React from 'react';
import Register from '../../../src/component/authentication/Register';
import renderer from 'react-test-renderer';

test('Register renders without crashing', () => {
  const tree = renderer.create(<Register />);
  expect(tree.toJSON()).toMatchSnapshot();
});
