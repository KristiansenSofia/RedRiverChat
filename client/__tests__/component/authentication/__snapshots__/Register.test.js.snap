// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestRegister renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionSticky-4 MuiAppBar-colorPrimary-7"
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <div
      className="MuiToolbar-root-36 MuiToolbar-gutters-37"
    >
      <h2
        className="MuiTypography-root-38 MuiTypography-title-44 MuiTypography-colorInherit-57"
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <button
        aria-label="Menu"
        className="MuiButtonBase-root-68 MuiIconButton-root-62 MuiIconButton-colorInherit-63"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "color": "#f5f5f5",
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-67"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-71"
            color={undefined}
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-77"
        />
      </button>
    </div>
  </header>
  <div
    className="Body"
  />
</div>
`;
